### PR TITLE
Generate additionnal etymology-free dictionnaries

### PR DESCRIPTION
**Pull request by BoboTiG**
Related to #1149.

@lasconic that's quite a big PR :sweat_smile: 

Overall:
- Jinja2 for HTML word templates
- Word HTML code now tested for all supported dict types
- The `Word` "type" is now properly typed to catch any issue elsewhere
- Most of interesting code changes are into `convert.py`.
- Sorry for Mypy in tests... In the other hand, it 's done and future changes should still be easy to do.

### CODE REVIEW

**Comment by BoboTiG** [comment]
Actually, let me know if Mypy for tests is too much. I will revert it then.

File: tests/test_3_convert.py [hunk 1]
```diff
@@ -89,66 +140,115 @@ def test_no_json_file() -> None:
...
+        "20201218",
+        include_etymology=include_etymology,
+    )
     assert (output_dir / filename).is_file()
+
+
+FORMATTED_WORD_KOBO = """\
+<w><p><a name="Multiple Etymologies"/><b>Multiple Etymologies</b> pron <i>gender</i>.<br/><br/><p>etyl 1</p><ol><li>setyl 1</li></ol><br/><ol><li>def 1</li><ol style="list-style-type:lower-alpha"><li>sdef 1</li></ol></ol></p><var><variant name="multiple etymology"/></var></w>
+"""  # noqa
+FORMATTED_WORD_KOBO_NO_ETIMOLOGY = """\
```
Comment (lasconic) [inline tests/test_3_convert.py:203]: Should be "FORMATTED_WORD_KOBO_NO_ET*Y*MOLOGY"

**Comment by lasconic** [comment]
Looks good. I'm cool with mypy for tests. 

Do you prefer to push the dictionaries with etymologies instead of the other ones ? If not, could be better to have "with-etym" for the "old" ones.

**Comment by BoboTiG** [comment]
> Do you prefer to push the dictionaries with etymologies instead of the other ones ? If not, could be better to have "with-etym" for the "old" ones.

I would say we do not touch original dictionnaries: they should be the "full" version.

**Comment by BoboTiG** [comment]
OK, the CI works perfectly with new files to upload (cf https://github.com/BoboTiG/ebook-reader-dict/releases/tag/no).

@lasconic do you want to change something about default dictionaries, or is it OK to merge?

**Comment by lasconic** [comment]
OK

**Pull request merged**